packages/mobile/src/web3/saga.ts

```diff
@@ -212,9 +270,16 @@ export function* unlockAccount(account: string) {
     }
 
     const pincode = yield call(getPincode)
-    yield call(web3.eth.personal.unlockAccount, account, pincode, UNLOCK_DURATION)
-    Logger.debug(TAG + '@unlockAccount', `Account unlocked: ${account}`)
-    return true
+    if (isZeroSyncMode()) {
+      Logger.info(TAG + '@unlockAccount', `unlockDuration is ignored in Geth free mode`)
+      const privateKey: string = yield readPrivateKeyFromLocalDisk(account, pincode)
```

Comment:
So we re-read the file and re-add the account every time we unlock it in zeroSyncMode?

Reply:
Re-add is a no-operation so the code has no correctness issue.
Re-reading might appear inefficient but it isn't, we are re-reading from keystore (which is a file) as well.

A more efficient option would be to first check if the account is already added or not. `contractkit` has a better support for this. And I can backport that code from it to `walletkit` as well but it might be easier to clean up this code when we move to `contractkit`. I added a TODO.